File: case_management/case_management/report/matter_summary_report/matter_summary_report.py
# Copyright (c) 2013, masonarmani38@gmail.com and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
from frappe import _

import frappe


def execute(filters=None):
    return get_column(), get_data(filters)


def get_data(filters):
    conditions = "(1=1) "
    if filters.get('opened_from') and filters.get('opened_to'):
        conditions += " and open_date  BETWEEN '{0}' and '{1}'".format(
            filters.get("opened_from"), filters.get('opened_to'))

    if filters.get('status'):
        conditions += " and status = '{0}'".format(filters.get("status"))

    if filters.get('client'):
        conditions += " and client = '{0}'".format(filters.get("client"))

    if filters.get('matter'):
        conditions += " and name = '{0}'".format(filters.get("matter"))

    if filters.get('responsible_solicitor'):
        conditions += " and responsible_solicitor = '{0}'".format(filters.get("responsible_solicitor"))

    # open_date, matter, client , status , practice_area , responsible_solicitor, close_date
    sql = "select open_date, name, client , status , practice_area , transaction, responsible_solicitor from `tabMatter` WHERE {0}"
    data = frappe.db.sql(sql.format(conditions))
    return data


def get_column():
    # ["Link:Link/Accident:150", "Data:Data:200", "Currency:Currency:100", "Float:Float:100"]
    return [
        _("Open Date") + ":Data:100",
        "Matter:Link/Matter:130",
        "Client:Link/Customer:200",
        "Status:Data:70",
        "Practice Area:Link/Practice Area:100",
        "Transaction:Link/Practice Area:100",
        "Solicitor:Link/Employee:140"
	]

File: case_management/case_management/report/matter_summary_report/matter_summary_report.js
// Copyright (c) 2016, masonarmani38@gmail.com and contributors
// For license information, please see license.txt

frappe.query_reports["Matter Summary Report"] = {
    "filters": [
        {
            "fieldname": "opened_from",
            "label": __("Opened From"),
            "fieldtype": "Date",
            "width": "80",
            "reqd": 1,
            "default": dateutil.year_start()
        },
        {
            "fieldname": "opened_to",
            "label": __("Opened To"),
            "fieldtype": "Date",
            "width": "80",
            "reqd": 1,
            "default": dateutil.year_end()
        },
        {
            "fieldname": "responsible_solicitor",
            "label": __("Responsible Solicitor"),
            "fieldtype": "Link",
            "options":"Employee",
            "width": "80",
            "reqd": 0,
            "default": ""
        },
        {
            "fieldname": "matter",
            "label": __("Matter"),
            "fieldtype": "Link",
            "options":"Matter",
            "width": "80",
            "reqd": 0,
            "default": ""
        },
        {
            "fieldname": "client",
            "label": __("Customer"),
            "fieldtype": "Link",
            "options":"Customer",
            "width": "120",
            "reqd": 0,
            "default": ""
        }
    ]
}
